libmscore/score.cpp

```diff
@@ -4080,12 +4080,124 @@ ChordRest* Score::cmdNextPrevSystem(ChordRest* cr, bool next)
 
 Box* Score::cmdNextPrevFrame(MeasureBase* currentMeasureBase, bool next) const
       {
-            Box* selectedBox { nullptr };
-            while (!selectedBox && (currentMeasureBase = (next ? currentMeasureBase->next() : currentMeasureBase->prev()))) {
-                  if (currentMeasureBase->isBox())
-                        selectedBox = toBox(currentMeasureBase);
+      Box* selectedBox { nullptr };
+      while (!selectedBox && (currentMeasureBase = (next ? currentMeasureBase->next() : currentMeasureBase->prev()))) {
+            if (currentMeasureBase->isBox())
+                  selectedBox = toBox(currentMeasureBase);
+            }
+      return selectedBox;
+      }
+
+//---------------------------------------------------------
+//   cmdNextPrevSection
+//    Return [Box* or ChordRest*] of next/previous section
+//---------------------------------------------------------
+
+Element* Score::cmdNextPrevSection(Element* el, bool dir) const
+      {
+      auto currentMeasureBase = el->findMeasureBase();
+      auto destination = currentMeasureBase;
+      if (currentMeasureBase) {
+            // -----------------------
+            // Next Section of Score
+            // -----------------------
+            if (dir) {
+                  if ((destination = getNextPrevSectionBreak(currentMeasureBase, true))) {
+                        el = getScoreElementOfMeasureBase(destination->next());
+                        }
                   }
-            return selectedBox;
+            // -------------------------
+            // Previous Section of Score
+            // -------------------------
+            else {
+                  auto currentSegment = el->isChordRest() ? toChordRest(el)->segment() : nullptr;
+                  if ((destination = getNextPrevSectionBreak(currentMeasureBase, false))) {
+                        if (currentSegment) {
```

Comment:
Dealt with segments here so that 1) Previous Section acts like Previous System in that the user will go to the beginning of the current section first if "in the middle" of a section (even if in its first measure but not at its beginning)